editoast/src/models/infra.rs

```diff
@@ -150,23 +151,26 @@ impl Infra {
     }
 
     pub async fn clone(infra_id: i64, db_pool: Data<DbPool>, new_name: String) -> Result<Infra> {
-        let infra_to_clone = Infra::retrieve(db_pool.clone(), infra_id).await?.unwrap();
-
-        let mut conn = db_pool.get().await?;
-        sql_query(
-            "INSERT INTO infra (name, railjson_version, owner, version, generated_version, locked, created, modified
+        let infra_to_clone = Infra::retrieve(db_pool.clone(), infra_id).await?;
```

Comment:
```suggestion
        let infra_to_clone = Infra::retrieve(db_pool.clone(), infra_id).await?.ok_or(InfraError::NotFound(infra_id).into())?;
```

I didn’t test it, so it might not work as it, but the idea to flatten to avoid nesting the code inside the `if let`

But very nice to chase the unwraps to return actual errors!